### PR TITLE
getAccountInfoAndSubscribe, subscription-based-orderbook

### PR DESCRIPTION
**Pull request by loganjross**

### CODE REVIEW

**Comment by Tristyn** [comment]
Ideally you'd provide some helper method like `market.subscribeToOrderbook() instead of getting the frontend to do it`

**Review by Tristyn — DISMISSED** [review]

**Review by jetjake — COMMENTED** [review]
This change is making it more complicated for consumers to use our library, not easier.

File: libraries/ts/src/margin/serum/market.ts [hunk 1]
```diff
@@ -550,15 +550,10 @@ export class Market {
...
+   * @param asksBuffer
    */
-  async loadOrderbook(provider: AnchorProvider): Promise<Orderbook> {
-    const bidsBuffer = (await provider.connection.getAccountInfo(translateAddress(this.marketConfig.bids)))?.data
-    const asksBuffer = (await provider.connection.getAccountInfo(translateAddress(this.marketConfig.asks)))?.data
-    if (!bidsBuffer || !asksBuffer) {
-      throw new Error("Orderbook sides not found")
-    }
-
+  async loadOrderbook(bidsBuffer: Buffer, asksBuffer: Buffer): Promise<Orderbook> {
```
[the remark this answers is not in the document]
Comment (loganjross) [reply libraries/ts/src/margin/serum/market.ts:556]: @jetjake okie dokie, any suggestions?

[the remark this answers is not in the document]
Comment (loganjross) [reply libraries/ts/src/margin/serum/market.ts:556]: @jetjake do you consider instantiating a provider connection or fetching an account to be more work at the top?

[the remark this answers is not in the document]
Comment (loganjross) [reply libraries/ts/src/margin/serum/market.ts]: spoke in slack, found a way i may be able to revert this method anyways

**Pull request closed without merging**